src/Contracts/SearchQuery.php

```diff
@@ -4,6 +4,25 @@
 
 namespace Meilisearch\Contracts;
 
+class FederationOptions
+{
+    private ?float $weight = null;
+
+    public function setWeight(float $weight): FederationOptions
+    {
+        $this->weight = $weight;
+
+        return $this;
+    }
+
+    public function toArray(): array
+    {
+        return array_filter([
+            'weight' => $this->weight ?? null,
```

Comment:
```suggestion
            'weight' => $this->weight,
```
weight is always defined or null, so no need for null-coalsecing